core-bundle/tests/Mailer/AvailableTransportsTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Mailer;
+
+use Contao\CoreBundle\Mailer\AvailableTransports;
+use Contao\CoreBundle\Mailer\TransportConfig;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\CoreBundle\Tests\TestCase;
+use Doctrine\Common\Annotations\AnnotationReader;
+
+class AvailableTransportsTest extends TestCase
+{
+    public function testAnnotatedCallbacks(): void
+    {
+        $service = new AvailableTransports();
+
+        $annotationReader = new AnnotationReader();
+        $annotations = $annotationReader->getMethodAnnotations(new \ReflectionMethod($service, 'getTransportOptions'));
+
+        $pageCallback = new Callback();
+        $pageCallback->table = 'tl_page';
+        $pageCallback->target = 'fields.mailerTransport.options';
+
+        $formCallback = new Callback();
+        $formCallback->table = 'tl_form';
+        $formCallback->target = 'fields.mailerTransport.options';
+
+        $this->assertEquals(
+            [
+                $pageCallback,
+                $formCallback,
+            ],
+            $annotations
+        );
```

Comment:
You could do sth. like this (explicitly checking the parts):

```suggestion
        $this->assertCount(2, $annotations);
        [$pageCallback, $formCallback] = $annotations;

        $this->assertInstanceOf(Callback::class, $pageCallback);
        $this->assertSame(
            ['table' => 'tl_page', 'target' => 'fields.mailerTransport.options', 'priority' => null],
            get_object_vars($pageCallback)
        );

        $this->assertInstanceOf(Callback::class, $formCallback);
        $this->assertSame(
            ['table' => 'tl_form', 'target' => 'fields.mailerTransport.options', 'priority' => null],
            get_object_vars($formCallback)
        );
```

Reply:
Oh, I just saw that's exactly what you did in 94203d730ba2650b314e35cdbd063e631fcd8e31. :smile:

Reply:
Yeah, although I used an array cast instead.

Reply:
Yours is much better, updated ;)